earth2studio/models/px/dlesym.py

```diff
@@ -0,0 +1,1077 @@
+
+        self.atmos_input_times = atmos_input_times
+        self.ocean_input_times = ocean_input_times
+        self.full_input_times = np.arange(
+            self.ocean_input_times[0],
+            self.atmos_input_times[-1] + 1,
+            atmos_input_times[1] - atmos_input_times[0],
+        )
+        self.atmos_output_times = atmos_output_times
+        self.ocean_output_times = ocean_output_times
+
+        # Setup the insolation and coupling times from the input and output times
+        self.atmos_sol_times = np.concatenate(
+            [self.atmos_input_times, self.atmos_output_times]
+        )
+        self.ocean_sol_times = np.concatenate(
+            [self.ocean_input_times, self.ocean_output_times]
+        )
+        n_atmos_steps = 1 + max(
+            self.atmos_model.output_time_dim // self.atmos_model.input_time_dim, 1
+        )
+        self.atmos_coupling_times = np.array(
+            [self.atmos_input_times[-1]] * n_atmos_steps
+        )
+        self.ocean_coupling_times = self.atmos_output_times
+
+    def input_coords(self) -> CoordSystem:
+        """Input coordinate system of the prognostic model
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        return OrderedDict(
+            {
+                "batch": np.empty(0),
+                "time": np.empty(0),
+                "lead_time": self.full_input_times,
+                "variable": np.array(self.atmos_variables + self.ocean_variables),
+                "face": np.arange(12),
+                "height": np.arange(self.nside),
+                "width": np.arange(self.nside),
+            }
+        )
+
+    @batch_coords()
+    def output_coords(self, input_coords: CoordSystem) -> CoordSystem:
+        """Output coordinate system of the prognostic model
+
+        Parameters
+        ----------
+        input_coords : CoordSystem
+            Input coordinate system to transform into output_coords
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+
+        output_coords = OrderedDict(
+            {
+                "batch": np.empty(0),
+                "time": np.empty(0),
+                "lead_time": self.atmos_output_times,
+                "variable": np.array(self.atmos_variables + self.ocean_variables),
+                "face": np.arange(12),
+                "height": np.arange(self.nside),
+                "width": np.arange(self.nside),
+            }
+        )
+
+        output_coords["batch"] = input_coords["batch"]
+        output_coords["time"] = input_coords["time"]
+        output_coords["lead_time"] = (
+            input_coords["lead_time"][-1] + output_coords["lead_time"]
+        )
+
+        return output_coords
+
+    @classmethod
+    def load_default_package(cls) -> Package:
+        """Default DLESyM model package on NGC"""
+        # TODO use NGC package when ready
+        return Package(
```

Comment:
Just adding comment to resolve before merge

Reply:
Update to throw a meaningful NotImplemented error maybe saying the package will be getting posted soon.